src/components/Dashboard/RewardsModule.js

```diff
@@ -85,18 +113,28 @@ const RewardsModule = React.memo(function RewardsModule({
       }
 
       // Withdraw funds from treasury
-      rewardTransactionQueue.push(
-        await onWithdraw(feeToken.id, wallet.account, totalFees)
-      )
+      if (totalTreasuryFees.gt(0)) {
+        rewardTransactionQueue.push(
+          await onWithdraw(feeToken.id, wallet.account, totalTreasuryFees)
+        )
+      }
+
+      // Claim subscription fees
+      for (const subscriptionFee of subscriptionFees) {
+        rewardTransactionQueue.push(
+          await onClaimSubscriptionFees(subscriptionFee.periodId)
+        )
+      }
 
       await Promise.all(rewardTransactionQueue.map(tx => tx.wait()))
+
+      setSubscriptionFees([])
```

Comment:
Yes this is definitely a possible scenario. 

>  But once their transactions were mined, and the subgraph updates, we would be able to start from n+1 (in the example, the third transaction)?

And yes this is what would happen in the case that we have rewards that are not subscriptions.

For court subscriptions, since we are fetching the data once on page load for now, in the scenario described above the page would need to be reloaded again. Once we have https://github.com/aragon/court-dashboard/pull/204#discussion_r395118931 implemented this shouldn't be a problem.